Resources/Catalog/RucioFileCatalogClient.py

```diff
@@ -0,0 +1,590 @@
+                               'GUID': guid,
+                               'Mode': 436,
+                               'ModificationDate': meta['updated_at'],
+                               'NumberOfLinks': 1,
+                               'Size': meta['bytes'],
+                               'Status': '-'}
+            try:
+              listFiles.remove(lfn)
+            except ValueError:
+              pass
+      except DataIdentifierNotFound as err:
+        failed[lfn] = str(err)
+      except Exception as err:
+        return S_ERROR(str(err))
+    for lfn in listFiles:
+      failed[lfn] = 'No such file or directory'
+    resDict = {'Failed': failed, 'Successful': successful}
+    return S_OK(resDict)
+
+
+  @checkCatalogArguments
+  def exists(self, lfns):
+    """ Check if the path exists
+    """
+    result = {'OK': True, 'Value': {'Successful': {}, 'Failed': {}}}
+    for lfn in lfns:
+      try:
+        did = self.__getDidsFromLfn(lfn)
+        exists = True
+        self.client.get_metadata(did['scope'], did['name'])
+      except DataIdentifierNotFound:
+        exists = False
+      except Exception as err:
+        return S_ERROR(str(err))
+      result['Value']['Successful'][lfn] = exists
+    return result
+
+
+  @checkCatalogArguments
+  def getFileSize(self, lfns):
+    """ Get the size of a supplied file
+    """
+    result = {'OK': True, 'Value': {'Successful': {}, 'Failed': {}}}
+    for lfn in lfns:
+      try:
+        did = self.__getDidsFromLfn(lfn)
+        meta = self.client.get_metadata(did['scope'], did['name'])
+        if meta['did_type'] == 'FILE':
+          result['Value']['Successful'][lfn] = meta['bytes']
+        else:
+          result['Value']['Successful'][lfn] = 0
+      except DataIdentifierNotFound:
+        result['Value']['Failed'][lfn] = 'No such file or directory'
+      except Exception as err:
+        return S_ERROR(str(err))
+    return result
+
+
+
+  @checkCatalogArguments
+  def isDirectory(self, lfns):
+    """ Determine whether the path is a directory
+    """
+    result = {'OK': True, 'Value': {'Successful': {}, 'Failed': {}}}
+    dids = []
+    for lfn in lfns:
+      dids.append(self.__getDidsFromLfn(lfn))
+    try:
+      for meta in self.client.get_metadata_bulk(dids):
+        lfn = str(meta['name'])
+        if meta['did_type'] in ['DATASET', 'CONTAINER']:
+          result['Value']['Successful'][lfn] = True
+        else:
+          result['Value']['Successful'][lfn] = False
+      for lfn in lfns:
+        if lfn not in result['Value']['Successful'] and lfn not in result['Value']['Failed']:
+          result['Value']['Failed'][lfn] = 'No such file or directory'
+    except Exception as err:
+      return S_ERROR(str(err))
+    return result
+
+
+
+  @checkCatalogArguments
+  def isFile(self, lfns):
+    """ Determine whether the path is a file
+    """
+
+    result = {'OK': True, 'Value': {'Successful': {}, 'Failed': {}}}
```

Comment:
And then return S_OK?
```suggestion
    result = {'Successful': {}, 'Failed': {}}
```

Reply:
Done